src/main/scala/uk/gov/nationalarchives/Lambda.scala

```diff
@@ -17,11 +17,13 @@ import fs2.interop.reactivestreams.StreamOps
 import software.amazon.awssdk.transfer.s3.model.CompletedUpload
 
 import java.io.{InputStream, OutputStream}
+import java.time.OffsetDateTime
 import java.util.UUID
 
 class Lambda extends RequestStreamHandler {
 
-  private val xmlCreator: XMLCreator = XMLCreator()
+  val ingestDateTime: () => OffsetDateTime = () => OffsetDateTime.now()
+  private lazy val xmlCreator: XMLCreator = XMLCreator(ingestDateTime)
```

Comment:
You've got a default value in the `apply` method which defaults to `.now()` so can you just have `XMLCreator()`?

Reply:
Ah, I was supposed to remove the default as I need to pass in the date for testing purposes.